src/transformutil.ts

```diff
@@ -172,20 +173,151 @@ async function dataToJson(file_data: string, opt: any = null) {
+                        switch (ssnode.nodeName) {
+                            case 'time':
+                                trkpt.time = new Date(ssnode.childNodes[0].data)
+                                break
+                            case 'ele':
+                                trkpt.ele = parseFloat(ssnode.childNodes[0].data)
+                                break
+                            case 'extensions':
+                                var extNodes = ssnode.childNodes
+                                for (
+                                    var idxExtNode = 0;
+                                    idxExtNode < extNodes.length;
+                                    idxExtNode++
+                                ) {
+                                    var extNode = extNodes[idxExtNode]
+                                    //console.log(extNode.nodeName)
+                                    if (extNode.nodeName == 'gpxtpx:TrackPointExtension') {
+                                        //console.log(extNode)
+                                        var trackPointNodes = extNode.childNodes
+                                        for (
+                                            var idxTrackPointNode = 0;
+                                            idxTrackPointNode < trackPointNodes.length;
+                                            idxTrackPointNode++
+                                        ) {
+                                            var trackPointNode =
+                                                trackPointNodes[idxTrackPointNode]
+                                            //console.log(trackPointNode.nodeName)
+                                            if (trackPointNode.nodeName.startsWith('gpxtpx:')) {
+                                                var gpxName = trackPointNode.nodeName.split(':')
+
+                                                trkpt[gpxName[1]] =
+                                                    trackPointNode.childNodes[0].data
+                                            }
+                                        }
+                                    }
+                                }
+                                //console.log(ssnode.childNodes)
+                                //extNode.forEach(element => {
+                                //console.log(element.power)
+                                //})
+                                break
+                        }
+                    }
+                    //console.log("trkpt", trkpt)
+                    segment.push(trkpt)
+                }
+            }
+            break
+    }
+    let len = node && node.childNodes && node.childNodes.length
+    for (
+        var idxChildNodes = 0;
+        idxChildNodes < len;
+        idxChildNodes++
+    ) {
+        getGpxData(node.childNodes[idxChildNodes], result)
+    }
+    return result
+}
+
+
+export async function trasformGpxData(data: any) {
```

Comment:
this function is used locally, can we remove export. also can we rename this function to trasformGpxDataToGeoJson to make it more meaningful

Reply:
fixed by roberto-butti in latest push